AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -437,6 +440,24 @@ - (void)setPlayerState:(ASVideoNodePlayerState)playerState
   _playerState = playerState;
 }
 
+- (void)setAssetURL:(NSURL *)assetURL
+{
+  if (!ASObjectIsEqual(assetURL, self.assetURL)) {
```

Comment:
Although it is probably not mandatory because the two methods (-assetURL, -setAsset:) each grab the lock - @Xinchi we should probably add a mutexlocker here because otherwise it is possible that in between the other two lock acquisitions, it could be yielded and that may be an issue with the "transactional" nature of an important write operation like this.